src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,801 @@
+
+            formatPath = System.IO.PathInternal.IsDeviceUNC(path.AsSpan()) ? @"\??" : @"\??\UNC\{0}\{1}";
+            uniqueId = Guid.NewGuid().ToString("N");
+            return string.Format(CultureInfo.InvariantCulture, formatPath, path.Trim('\\'), uniqueId);
+        }
+
+        #endregion
+
+    }
+
+    //-------------------------------------------------------------------------
+    // FileFullEaInformation
+    //
+    // Description: this class encapsulates the marshalling of data from a
+    // managed representation of the FILE_FULL_EA_INFORMATION struct into 
+    // native code. As part of this task, it manages memory that is allocated 
+    // in the native heap into which the managed representation is blitted. 
+    // The class also implements a SafeHandle pattern to ensure that memory
+    // is not leaked in "exceptional" circumstances such as Thread.Abort().
+    //
+    //-------------------------------------------------------------------------
+
+    internal class FileFullEaInformation : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        private string _eA_NAME_STRING = "Filestream_Transaction_Tag";
+        private int _m_cbBuffer;
+
+        public FileFullEaInformation(byte[] transactionContext)
+            : base(true)
+        {
+            _m_cbBuffer = 0;
+            InitializeEaBuffer(transactionContext);
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            _m_cbBuffer = 0;
+
+            if (base.handle == IntPtr.Zero)
+                return true;
+
+            Marshal.FreeHGlobal(base.handle);
+            base.handle = IntPtr.Zero;
+
+            return true;
+        }
+
+        public int Length
+        {
+            get
+            {
+                return _m_cbBuffer;
+            }
+        }
+
+        private void InitializeEaBuffer(byte[] transactionContext)
+        {
+            if (transactionContext.Length >= ushort.MaxValue)
+                throw ADP.ArgumentOutOfRange("transactionContext");
+
+            Interop.NtDll.FILE_FULL_EA_INFORMATION eaBuffer;
+            eaBuffer.NextEntryOffset = 0;
+            eaBuffer.Flags = 0;
+            eaBuffer.EaName = 0;
+
+            // string will be written as ANSI chars, so Length == ByteLength in this case
+            eaBuffer.EaNameLength = (byte)_eA_NAME_STRING.Length;
+            eaBuffer.EaValueLength = (ushort)transactionContext.Length;
+
+            // allocate sufficient memory to contain the FILE_FULL_EA_INFORMATION struct and
+            // the contiguous name/value pair in eaName (note: since the struct already
+            // contains one byte for eaName, we don't need to allocate a byte for the 
+            // null character separator).
+            _m_cbBuffer = Marshal.SizeOf(eaBuffer) + eaBuffer.EaNameLength + eaBuffer.EaValueLength;
+
+            IntPtr pbBuffer = IntPtr.Zero;
+            RuntimeHelpers.PrepareConstrainedRegions();
+            try
```

Comment:
We could do away with the try{}finally{} block and have the code in the finally block as-is. 
There are no thread abort in .Net Core.